FirebaseVertexAI/Sources/GenerativeAIService.swift

```diff
@@ -146,13 +151,24 @@ struct GenerativeAIService {
 
   // MARK: - Private Helpers
 
-  private func urlRequest<T: GenerativeAIRequest>(request: T) throws -> URLRequest {
+  private func urlRequest<T: GenerativeAIRequest>(request: T) async throws -> URLRequest {
     var urlRequest = URLRequest(url: request.url)
     urlRequest.httpMethod = "POST"
     urlRequest.setValue(apiKey, forHTTPHeaderField: "x-goog-api-key")
-    urlRequest.setValue("genai-swift/\(GenerativeAISwift.version)",
-                        forHTTPHeaderField: "x-goog-api-client")
+    // TODO: Determine the right client header to use.
+    //    urlRequest.setValue("genai-swift/\(GenerativeAISwift.version))",
```

Comment:
FirebaseVersion() ?

Reply:
I think `FirebaseVersion()` will be what we want instead of `GenerativeAISwift.version` but the prefix `genai-swift` will need to change too. Will leave it all commented out until that part is figured out.